### PR TITLE
Add support for Subscription Schedule APIs

### PR DESCRIPTION
**Pull request by remi-stripe**
Add support for Subscription Schedule APIs

* Add Subscription Schedule
* Add Subscription Schedule Revisions
* Add `invoice_settings` on the Customer

cc @stripe/api-libraries

### CODE REVIEW

**Comment by remi-stripe** [comment]
This is now ready to be reviewed. r? @mickjermsurawong-stripe

File: src/main/java/com/stripe/model/SubscriptionSchedule.java [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+  }
+
+  public void setSubscriptionObject(Subscription expandableObject) {
+    this.subscription =
+        new ExpandableField<Subscription>(expandableObject.getId(), expandableObject);
+  }
+
+  // <editor-fold desc="cancel">
+  /**
+   * Pay a subscription schedule.
```
Comment (mickjermsurawong-stripe) [inline src/main/java/com/stripe/model/SubscriptionSchedule.java]: Should be cancel a subscription schedule?

Comment (remi-stripe) [reply src/main/java/com/stripe/model/SubscriptionSchedule.java]: damn copy/paste and replace. Fixed all 8 of them.

File: src/main/java/com/stripe/model/SubscriptionScheduleRevisionCollection.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package com.stripe.model;
+
+import com.stripe.Stripe;
+import com.stripe.exception.StripeException;
+import com.stripe.net.ApiResource;
+import com.stripe.net.RequestOptions;
+import java.util.Map;
+
+public class SubscriptionScheduleRevisionCollection
```
Comment (mickjermsurawong-stripe) [inline src/main/java/com/stripe/model/SubscriptionScheduleRevisionCollection.java]: Hmm.. actually no other resource returns this collection, so we can never get an instance of this collection to make the API request (with the existing url)..

Comment (mickjermsurawong-stripe) [reply src/main/java/com/stripe/model/SubscriptionScheduleRevisionCollection.java]: I think we can go with the followings..
In `SubscriptionSchedule.java` we have
`public SubscriptionScheduleRevisionCollection revisions() `
Then, we remove the `list` method from the `SubscriptionScheduleRevisionCollection.java`..
I fixed my client metadata and the generated outcome looks like this..
https://github.com/stripe/stripe-java/blob/abaada5f48449f495886efdda0eed0c5583bbe0b/src/main/java/com/stripe/model/SubscriptionSchedule.java#L338-L356


I believe this is similar to account/person relationship

**Comment by mickjermsurawong-stripe** [comment]
ptal @remi-stripe just on the revision collection because it is currently not returned anywhere..

File: src/main/java/com/stripe/model/SubscriptionScheduleRevisionCollection.java (hunk 2 above)
Comment (remi-stripe) [reply src/main/java/com/stripe/model/SubscriptionScheduleRevisionCollection.java]: @mickjermsurawong-stripe Nice catch, fixed!

**Comment by mickjermsurawong-stripe** [comment]
thank @remi-stripe 
LGTM

**Pull request merged**